src/products/dto/product.dto.ts

```diff
@@ -0,0 +1,31 @@
+import { IsNotEmpty, IsNumber, IsOptional, IsString } from "class-validator";
+
+export class ProductDto {
+    @IsNotEmpty()
+    @IsString() 
+    name: string; 
+    
+    @IsNotEmpty()
+    @IsNumber() 
+    price: number;
+
+    @IsNotEmpty()
+    @IsString() 
+    barcode: string; 
+   
+    @IsNotEmpty()
+    @IsNumber() 
+    category_id: number
```

Comment:
Naming Convention: I prefer using categoryId instead of category_id. 
https://github.com/felixge/node-style-guide#use-lowercamelcase-for-variables-properties-and-function-names

Reply:
I did it like this just to be able to work with the Angular ui